### ISSUE TITLE
[Cutscenes] Opening animation?

### ISSUE THREAD

**Issue by LMitch83** [ribbybibby/Grimm-Journey#43]
While playing the latest build, I was wondering whether at some point we should have a little comic-book style animation to open the game (much like the end one we're planning).

I'm thinking a series of comic-book panels following the existing rhyming narrative; this (and the narrative) would play after the 'press any key' but before the game itself, and can be skipped by the player.

Thoughts?

**Comment by HoboMechanistic** [comment]
You took the words right out of my mouth. I was going to log this as a potential enhancement. I am planning to look into how to add video to Unity / see how Unity processes video files. I feel it would be easier to tackle this that way rather than trying to set something up inside Unity.

Either way, I think this would be a super cool thing to have now that we have neat voice overs.

**Comment by HoboMechanistic** [comment]
Looks like this is another case of "Need Unity Pro". Second time we ran into this wall, I guess we are getting good at this game dev thing :D 

Unity Pro is required as video files in game are played via a render texture that you attach to game objects (like how we attach textures to quads for example).

There is a work around for this however, as other people have used this in the past. What we could do is make the video in some video creating software. After this, use some tool (many others have used Virtual Dub) to split the video up into its frame components.

We then import all the frames and write some code to play the frames at 30 FPS. So basically its a half video import / half unity code jobby. In terms of size, one of the cases I read up on said his video was 5 mb in size and the total amount of images after being divided was 6 mb. So we shouldn't bloat anything by trying this. If you, Laura and Stace want to work on making this the code side shouldn't be to difficult to implement.

So in short - we can use a ready made video, permitting we can divide the video into its separate frames.

**Comment by LMitch83** [comment]
Message to Stace - when you're ready, I would be interested to hear your thoughts on this. I'm now considering the best plan would be to keep everything in the shadow-puppet world (including opening and end scene animations), but have the animation to be a bit minimalist (like those Flash animations where just key elements would move, such as an arm raising or a head turning) so they still kind of look like puppets. Rob DJ says you've been looking at some new animation software, so was curious as to your take on this.

**Comment by frozenhowl** [comment]
So as far as I can tell theres maybe 2 ways we can do this.
1. Using all our in game assests and animations and make a cutscene that way using minamilist movements (would be done using the spine software and imported)
2. Make a slightly more artistic version so it clearly looks like a cutscene - im picturing along the lines of the 3 Brothers tale from Harry Potter - https://www.youtube.com/watch?v=bN1_h_eGitE
so it keeps the shadow puppet  style but looks different from the game. (probably done with sony vegas maybe or other software)

I guess it might just come down to how are we telling the story - is it all meant to be played out in a shadow puppet theater or by other means etc. so we want to have fancy cutscenes or ingame stuff :)

**Comment by HoboMechanistic** [comment]
Loving that video :D Once I get us the Spine software we can look at what is possible. Personally, I would like to have a difference between in-game and cutscenes? This is up to you two however. 

We could still use the same assets but maybe in more detail for cutscenes? Once we have the software and start importing things into the engine we can see what is possible :)

**Comment by LMitch83** [comment]
Thought I'd replied to this (might have been via email?). Adding additional thoughts here anyway for posterity (and apologies if I hadn't replied in the first place :( ).

I like the style of the Three Brothers tale, but I do think we should try and stick to the shadow puppet theme, so that it plays out like a puppet theatre. However, if you have any interesting art/animation ideas that are different from the in-game assets but still keep this general tone, I'd be happy to see them. I don't mind variation, so long as we still keep the shadow puppet feel.

In terms of cutscenes, I thought we should keep it short and sweet. I'm thinking a pre-title screen cutscene and an ending cutscene of a couple of minutes length max, and a short cutscene using in-game assets to segue from one world to the other (so two in total as it stands).

I did start thinking about a pre-title-screen cutscene. I'm still working on the rhyme, but here's the basic plot. I'll add this to the wiki once I've got feedback:
BBW watches LRRH skip through a glad towards the eerie woods.
BBW's POV as he hunts down LRRH.
BBW strikes LRRH... but she swiftly pulls a bracelet out of her basket and slams it onto BBW's wrist.
BBW tries to attack LRRH, but recoils. LRRH is smug, BBW appears furious.
A eerie wind and rush of animals scurry past them in fear.
Suddenly, they see rolling darkness rushing towards them.
LRRH jumps up to a tree branch. BBW tries to jump, but cannot. 
The darkness rushed towards BBW.
LRRH jumps down from her branch and grabs BBW - the bracelet glows, and they both jump to safety.
The darkness rolls by. BBW and LRRH look at each other, then towards the eerie woods.
END.
